### PR TITLE
[FIX] Apply windows batch size

### PR DESCRIPTION
**Pull request by marcopeix**
Applies `windows_batch_size` before permuting to avoid materializing many windows into memory and running into OOM issues. 

With this change we fix issues #1357 and #1297.

This PR also replaces #1421; that PR will not be needed and this is a simpler fix.

### CODE REVIEW

File: neuralforecast/common/_base_model.py [hunk 1]
```diff
@@ -722,6 +722,12 @@ def _create_windows(self, batch, step):
                 dimension=-1, size=window_size, step=self.step_size
             )
 
+            # Sample windows before permute
```
Comment (elephaint) [inline neuralforecast/common/_base_model.py]: Shouldn't we delete the existing sampling method then?

File: neuralforecast/common/_base_model.py [hunk 2]
```diff
@@ -722,6 +722,12 @@ def _create_windows(self, batch, step):
...
+            # windows shape: [n_series, C, Ws, window_size]
+            if self.windows_batch_size is not None and windows.shape[2] > self.windows_batch_size:
```
Comment (elephaint) [inline neuralforecast/common/_base_model.py]: We can't sample before the creation of the windows because the available conditions establish which windows are eligible to sample from

**Review by elephaint — CHANGES_REQUESTED** [review]
I don't think this works; the sampling should be after we've established the eligible windows (otherwise we are sampling highly inefficiently).

Where are the windows materialized? What line of code? That's where the issue is.

**Comment by elephaint** [comment]
Edit: doesn't work - still same error 😞

File: neuralforecast/common/_base_model.py [hunk 3]
```diff
@@ -742,37 +742,58 @@ def _create_windows(self, batch, step):
...
-                    outsample_condition, axis=(1, -1)
-                )  # Sum over time & series dimension
-                final_condition = (outsample_condition >= min_outsample_points) & (
-                    insample_condition >= min_insample_points
-                )
+            
+            if self.MULTIVARIATE:
+                # For multivariate: windows shape is [Ws, L + h, C, n_series]
+                insample_condition = windows[:, : self.input_size, available_idx, :]
+                insample_condition = torch.sum(insample_condition, axis=(1, 2))
```
Comment (elephaint) [inline neuralforecast/common/_base_model.py]: The summing axis should be over the time and series dimension
